serde_codegen_internals/src/attr.rs

```diff
@@ -746,6 +785,64 @@ fn get_ser_and_de<T, F>(cx: &Ctxt,
     Ok((ser_item.get(), de_item.get()))
 }
 
+type TypeConversions<T> = (Option<T>, Option<T>, Option<T>, Option<T>);
+
+fn get_from_into_as(cx: &Ctxt,
```

Comment:
Add compile-fail tests for all of these error messages in test_suite/tests/compile-fail.